### PR TITLE
Display actual name of source in Music Library when sorted by source

### PR DESCRIPTION
**Pull request by poliwhirlge**
e.g. rb4 -> Rock Band 4

### CODE REVIEW

File: Assets/Script/Menu/MusicLibrary/MusicLibraryMenu.cs [hunk 1]
```diff
@@ -161,7 +161,8 @@ protected override List<ViewType> CreateViewList()
             foreach (var section in _sortedSongs)
             {
                 // Create header
-                list.Add(new SortHeaderViewType(section.Key, section.Value.Count));
+                string key = Sort == SongAttribute.Source ? SongSources.SourceToGameName(section.Key) : section.Key;
+                list.Add(new SortHeaderViewType(key, section.Value.Count));
```
Comment (sonicfind) [inline Assets/Script/Menu/MusicLibrary/MusicLibraryMenu.cs:165]: There's a small flaw that comes with this.
While this works well with sources defined in our database, anything that isn't appears as Custom/Unknown.

Which leads to stuff like this
image

The only way to address that issue in an efficient manner would be to have a second version of `SongSources.SourceToGameName` return whether or not a name was found for the given source alongside said name. 

If one was not found, then we could just append the inner source name to the end of `Custom/Unknown`.

I do agree with the basic idea though.

**Review by sonicfind — CHANGES_REQUESTED** [review]

File: Assets/Script/Menu/MusicLibrary/MusicLibraryMenu.cs (hunk 1 above)
Comment (sonicfind) [reply Assets/Script/Menu/MusicLibrary/MusicLibraryMenu.cs:165]: Making a second version of SongSources.SourceToGameName (preferably like `public static bool TrySourceToGameName(string id, out string name)`), will also require an accompanying `SongSources.GetSource` as well.

**Comment by poliwhirlge** [comment]
If ```GetSource(id)``` is already returning the backup source wouldn't that mean that we can't identify the source and thus can only append the id to the end of ```Custom/Unknown```? Or is there a field I've missed that we can use to fill in this information?

**Comment by sonicfind** [comment]
That's why I mentioned you need an accompanying `SongSources.TryGetSource` as well

**Review by sonicfind — APPROVED** [review]

**Pull request merged**